capa/features/freeze/__init__.py

```diff
@@ -135,6 +154,36 @@ class FileFeature(HashableModel):
     feature: Feature
 
 
+class ProcessFeature(HashableModel):
+    """
+    args:
+        process: the address of the process to which this feature belongs.
+        address: the address at which this feature is found.
+
+    process != address because, e.g., the feature may be found *within* the scope (process).
+    versus right at its starting address.
```

Comment:
i don't understand this - doesn't the `address` contain the information needed to pick the right process? can you maybe include an example that shows how these are not the same?

Reply:
isn't `address` the address returned by a feature? if so, the the environment strings extracted return a `NO_ADDRESS` for the moment (and so should the future registry and filename features I believe).
If I'm correct, then should we make the address of those features the same as the process address?

Reply:
yes, each feature is associated with an address.

when possible, we should associate features with the most specific possible address, like an API call feature associated with a DynamicAnalysisEventAddress (or similar). we should use ProcessAddress whenever the feature can only be related to a process but we're not sure at what specific point. an environment string is a good example of this - the string is related to some process, not a specific event. 

we should use `NO_ADDRESS` when we really have no relation to any specific object. im not sure if there are any good examples of this yet for dynamic analysis. perhaps CAPE version (but this isn't a feature...).

Reply:
> versus right at its starting address.

to clarify, i dont think we should consider ProcessAddress to refer to a specific point in time, like the event that spawns a process or the moment it springs into existence; rather, it should represent the entire Process over time.